### DISCUSSION TITLE
Parallel using Key order

### DISCUSSION THREAD

**Discussion by spoon99** [confluentinc/parallel-consumer#774] [Q&A] [answered]
Hi, 
I'm trying to understand how key order is working. what I understand is it would be grouped by key (bucketed) and it would run parallel to the grouped keys. If I understood correctly, if the key variety is less than the parallelism number how would it work?

For example, I'm running PC with 6 parallelism, but all the incoming messages will have the same message key, Could it run with 6 parallelism, or would it just run with 1 thread?

**Comment by rkolesnev** [comment, marked as answer]
Hi @spoon99  - if all messages on the partition are with same key - in Key ordering mode Parallel Consumer will effectively run with parallelisation of 1 as to maintain per Key ordering of message processing.
